app/services/katello/rhsm_fact_parser.rb

```diff
@@ -55,7 +55,13 @@ def operatingsystem
       return nil if name.nil? || version.nil?
 
       os_name = distribution_to_puppet_os(name)
-      major, minor = version.split('.')
+      version_splits = version.split('.')
+      if name =~ /ubuntu/i
```

Comment:
Same comment about `.match?` as above. Or we already have a normalized name so `if os_name == 'Ubuntu'` should also be reliable.